content/Rust-1.87.0.md

```diff
@@ -0,0 +1,252 @@
++++
+path = "2025/05/15/Rust-1.87.0"
+title = "Announcing Rust 1.87.0"
+authors = ["The Rust Release Team"]
+aliases = [
+    "2025/05/15/Rust-1.87.0.html",
+    "releases/1.87.0",
+]
+
+[extra]
+release = true
++++
+
+The Rust team is happy to announce a new version of Rust, 1.87.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.87.0 with:
+
+```
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.87.0](https://doc.rust-lang.org/stable/releases.html#version-1870-2025-04-03).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.87.0 stable
+
+### Anonymous pipes
+
+1.87 adds access to anonymous pipes to the standard library. This includes
+integration with `std::process::Command`'s input/output methods.
+
+As a toy example, this program uses pipes to stream data into `cat` and read the output:
```

Comment:
Rather than a toy example, maybe it's worth putting a practical program people might actually use? I wanted the ability to combine captured stdout and stderr for so long, and I'm just excited it's now finally possible :D

```rust
use std::process::Command;
use std::io::Read;

let (mut recv, send) = std::io::pipe()?;

let mut command = Command::new("path/to/bin")
    // Both stdout and stderr will write to the same pipe, combining the two.
    .stdout(send.try_clone()?)
    .stderr(send)
    .spawn()?;

let mut output = Vec::new();
recv.read_to_end(&mut output)?;

// It's important that we read from the pipe before the process exits, to avoid
// filling the OS buffers if the program emits too much output.
assert!(command.wait()?.success());
```

Reply:
Yeah, that seems like a good idea. Not sure what a good candidate program is...

Reply:
You mean instead of `path/to/bin`? I think it'd be ok to leave the placeholder here, the goal I have here is not to have something people can copy/paste in the playground, but rather show that it's now possible to merge stdout and stderr, and show how people can do that in their projects.

Reply:
Ack, that makes sense. Pushed an update.